geonode/urls.py

```diff
@@ -65,6 +65,23 @@
     url(r"^messages/", include(msg_urls)),
 ]
 
+# custom_metadata
+if settings.EXTRA_METADATA_ENABLED:
```

Comment:
I would evaluate letting the external contrib app injects the URLs patterns needed for the following reasons:
- We are introducing in the core a concept that could be not part of it. This means that for each contrib app, the core should be aware of the URLs available (that is what we want to avoid)
- It may be more dynamic and safe to let the contrib app inject its own URL like (for example) we did for the [geonode importer](https://github.com/geosolutions-it/geonode-importer/blob/4911c526041d4887b7ba82955f73d1ef258bb5db/importer/apps.py#L22-L34). In this way any changes to the URLs (adding, delete, update) does not require a change in the core